client/src/main/java/io/quarkus/vault/client/http/VaultHttpClient.java

```diff
@@ -0,0 +1,83 @@
+package io.quarkus.vault.client.http;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.net.ConnectException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.TimeoutException;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import io.quarkus.vault.client.VaultClientException;
+import io.quarkus.vault.client.common.VaultErrorResponse;
+import io.quarkus.vault.client.common.VaultRequest;
+import io.quarkus.vault.client.common.VaultRequestExecutor;
+import io.quarkus.vault.client.common.VaultResponse;
+import io.quarkus.vault.client.json.JsonMapping;
+import io.smallrye.mutiny.Uni;
+
+public abstract class VaultHttpClient implements VaultRequestExecutor, AutoCloseable {
```

Comment:
a `VaultHttpClient` is not a `VaultClient`?

Reply:
would there be a different name, avoiding the confusion?

Reply:
I'm not sure what you are suggesting we rename. Currently (I believe) they are named for exactly what they do. Examples?